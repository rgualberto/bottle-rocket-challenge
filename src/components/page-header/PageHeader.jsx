import React from 'react';
import PropTypes from 'prop-types';

const PageHeader = props => {
  const {
    Title,
    onBackClick,
    onMapClick
  } = props;

  return (
    <div className="page__header">
      <div className="page__header-contain">
        <h1 className="page__heading">{Title}</h1>

        {(onBackClick !== null && onBackClick !== undefined) &&
          <div className="page__header-action page__header-action--back">
            <button onClick={onBackClick} className="page__icon-button">
              {/* NOTE: Asset here doesn't match comps. Also would prefer svgs to normalize icons across app */}
              <img src="/assets/images/ic_webBack@2x.png" />
            </button>
          </div>

        }

        {(onMapClick !== null && onMapClick !== undefined) &&
          <div className="page__header-action page__header-action--map">
            <button onClick={onMapClick} className="page__icon-button">
              <img src="/assets/images/icon_map@2x.png" />
            </button>
          </div>
        }
      </div>
    </div>
  );
}

PageHeader.propTypes = {
  Title: PropTypes.string,

  onBackClick: PropTypes.func,
  onMapClick: PropTypes.func
};

export default PageHeader;
