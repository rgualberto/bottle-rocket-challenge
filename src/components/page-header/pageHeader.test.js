import React from 'react';
import renderer from 'react-test-renderer';
import {fireEvent, render} from '@testing-library/react';
import PageHeader from './PageHeader.jsx';

test('PageHeader blank snapshot', () => {
  const component = renderer.create(
    <PageHeader
      Title="Lunch Tyme"
      onBackClick={null}
      onMapClick={null}
    />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('PageHeader contains map action icon when function provided and no back action icon when function not provided', () => {
  const {container} = render(
    <PageHeader
      Title="Lunch Tyme"
      onBackClick={null}
      onMapClick={() => undefined}
    />
  );

  expect(container.querySelector(".page__header-action--map")).not.toBeNull();

  expect(container.querySelector(".page__header-action--back")).toBeNull();
});

// test('PageHeader blank snapshot', () => {
//   const component = renderer.create(
//     <PageHeader
//       Title="Lunch Tyme"
//       onBackClick={null}
//       onMapClick={null}
//     />
//   );
//
//   let tree = component.toJSON();
//   expect(tree).toMatchSnapshot();
// });
