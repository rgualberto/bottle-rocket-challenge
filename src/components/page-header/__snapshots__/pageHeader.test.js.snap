// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeader blank snapshot 1`] = `
<div
  className="page__header"
>
  <div
    className="page__header-contain"
  >
    <h1
      className="page__heading"
    >
      Lunch Tyme
    </h1>
  </div>
</div>
`;
